### PR TITLE
Adding POST for ASV and all the db entity changes for container VNF deployment

### PR DESCRIPTION
**Pull request by emanoelxavier**
This PR implements:
1. All the code needed to create an appliance software version (entity changes, services, unit tests, rest API). see https://github.com/opensecuritycontroller/community/blob/master/designs/containers/vnf-deployment/vnf-deployment.md#appliance-software-version 

2. All the DB entity changes related to DS and DAIS for K8s, see https://github.com/opensecuritycontroller/community/blob/master/designs/containers/vnf-deployment/vnf-deployment.md#osc-entities 

3.  Makes the ASV integer field using `Integer` rather than `int`. Those fields were already optional but when not provided we had an NPE. This fixes that problem.

Note: Remaining changes to create a k8s DS will come later in a different PR

### CODE REVIEW

File: osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java [hunk 1]
```diff
@@ -239,6 +239,8 @@ private static void performUpdateChain(int curDbVer, Statement stmt, EncryptionA
                 upgrade84to85(stmt);
             case 85:
                 upgrade85to86(stmt);
+            case 86:
+                upgrade86to87(stmt);
```
Comment (dmitryintel) [inline osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java]: I think we are clashing here. I also have 87.

File: osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java [hunk 2]
```diff
@@ -250,6 +252,31 @@ private static void performUpdateChain(int curDbVer, Statement stmt, EncryptionA
         }
     }
     
+    private static void upgrade86to87(Statement stmt) throws SQLException {
+        execSql(stmt, "alter table APPLIANCE_SOFTWARE_VERSION add column image_pull_secret_name varchar(255);");
```
Comment (dmitryintel) [inline osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java]: I have also seen some changes to the Virtual System

File: osc-domain/src/main/java/org/osc/core/broker/model/entities/appliance/DistributedApplianceInstance.java [hunk 3]
```diff
@@ -48,12 +48,12 @@
 
     @ManyToOne(fetch = FetchType.EAGER)
     @JoinColumn(name = "virtual_system_fk", nullable = false,
-            foreignKey = @ForeignKey(name = "FK_DAI_VIRTUAL_SYSTEM"))
+    foreignKey = @ForeignKey(name = "FK_DAI_VIRTUAL_SYSTEM"))
```
[the remark this answers is not in the document]
Comment (emanoelxavier) [reply osc-domain/src/main/java/org/osc/core/broker/model/entities/appliance/DistributedApplianceInstance.java]: I have my eclipse settings to auto-indent when I save. This is just following the settings which we should have in common I believe.

File: osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java (hunk 1 above)
Comment (emanoelxavier) [reply osc-server/src/main/java/org/osc/core/broker/util/db/upgrade/ReleaseUpgradeMgr.java]: Whoever does the merge after needs to handle the conflict as usual.

File: osc-domain/src/main/java/org/osc/core/broker/model/entities/appliance/DistributedApplianceInstance.java [hunk 4]
```diff
@@ -82,6 +82,13 @@
     @Column(name = "os_server_id")
     private String osServerId;
 
+    @Column(name = "external_id")
+    private String externalId;
+    @Column(name = "inspection_element_id")
+    private String inspectionElementId;
+    @Column(name = "inspection_element_parent_id")
+    private String inspectionElementParentId;
```
[the remark this answers is not in the document]
Comment (emanoelxavier) [reply osc-domain/src/main/java/org/osc/core/broker/model/entities/appliance/DistributedApplianceInstance.java]: addressed

**Comment by emanoelxavier** [comment]
The latest commit on this PR adds the remaining code to allow for a creation of a K8s deployment spec. See https://github.com/opensecuritycontroller/community/blob/master/designs/containers/vnf-deployment/vnf-deployment.md#deployment-spec

**Pull request merged**